### PR TITLE
Cleanup of tests

### PR DESCRIPTION
**Pull request by lolgear**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description

...
CocoaPods are removed from Tests.

### CODE REVIEW

**Comment by sushichop** [comment]
Looks good for me totally.
(I see `pod repo update --silent` was removed in .travis.yml, but I think it is OK because `pod setup --slient` was remained. And I know `pod setup` includes the same process of `pod repo update` internal.)

Then I think it had better merge this PR before #965 for preventing code conflicts.
Because this PR haven't removed linkage about Pods/Pods.xcodeproj form Framework/Lumberjack.xcworkspace, on the other hand, #965 have already removed it.

**Comment by lolgear** [comment]
@sushichop 
Is `pod setup` necessary for `pod lib lint`?

**Comment by lolgear** [comment]
@sushichop 

Also, I would like to rename Tests project into `Tests.xcodeproj`.
So, it is better to remove project linkage in this PR.

**Comment by lolgear** [comment]
Here is a checklist:

- [ ] Rename test project in `Tests.xcodeproj`
- [ ] Choose between current approach (Proxy) and inheritance in Basic logging tests.
- [ ] Cleanup DDSMocking.

**Comment by sushichop** [comment]
> Is `pod setup` necessary for `pod lib lint`?

Yes. `pod setup` is necessary.

**Comment by lolgear** [comment]
@sushichop 
Could it be simplified only to install `lint` utility?

**Comment by sushichop** [comment]
@lolgear 

> Also, I would like to rename Tests project into `Tests.xcodeproj`.
> So, it is better to remove project linkage in this PR.

No problem at least for me:)

**Comment by sushichop** [comment]
@lolgear 

> Could it be simplified only to install `lint` utility?

Unfortunately, No. It couldn't. When we exececute `pod lib lint`, we need `install cocoapods`.
Thanks for your effort. I think you try to simplify it as possible:)

**Comment by sushichop** [comment]
@lolgear 

> Is pod setup necessary for pod lib lint?

Oops. I misread the above sentence😅
No. `pod setup` is not necessary for `pod lib lint`.

File: Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme [hunk 1]
```diff
@@ -5,28 +5,11 @@
...
-               ReferencedContainer = "container:CocoaLumberjack Tests.xcodeproj">
-            </BuildableReference>
-         </BuildActionEntry>
-      </BuildActionEntries>
    </BuildAction>
    <TestAction
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      codeCoverageEnabled = "YES"
```
Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme]: Is this intentional?

Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme]: Is this intentional?

Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme]: Is this intentional?

Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme]: Is this intentional?

Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme]: Is this intentional?

File: Tests/Tests.xcodeproj/xcshareddata/xcschemes/iOS Tests.xcscheme [hunk 2]
```diff
@@ -5,28 +5,11 @@
...
-               ReferencedContainer = "container:CocoaLumberjack Tests.xcodeproj">
-            </BuildableReference>
-         </BuildActionEntry>
-      </BuildActionEntries>
    </BuildAction>
    <TestAction
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      codeCoverageEnabled = "YES"
```
Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/iOS Tests.xcscheme]: Same as above: intentional?

File: Tests/Tests.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -345,37 +312,41 @@
...
 				COMBINE_HIDPI_IMAGES = YES;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(DEVELOPER_FRAMEWORKS_DIR)",
 					"$(inherited)",
 				);
 				INFOPLIST_FILE = Info.plist;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/../Frameworks";
 				SUPPORTED_PLATFORMS = macosx;
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 4.2;
```
Comment (ffried) [inline Tests/Tests.xcodeproj/project.pbxproj]: Do we have xcconfig files for the Tests? If yes, I think these settings belong in the xcconfig file.

Comment (ffried) [inline Tests/Tests.xcodeproj/project.pbxproj]: Do we have xcconfig files for the tests? If so, shouldn't these settings be set there?

Comment (ffried) [inline Tests/Tests.xcodeproj/project.pbxproj]: Do we have xcconfig files for the tests? If so, shouldn't these settings be set there?

File: Tests/Tests.xcodeproj/xcshareddata/xcschemes/iOS Tests.xcscheme (hunk 2 above)
Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/iOS Tests.xcscheme]: Was this removed intentionally?

Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/iOS Tests.xcscheme]: Same as above: Was this removed intentionally?

File: Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme (hunk 1 above)
Comment (ffried) [inline Tests/Tests.xcodeproj/xcshareddata/xcschemes/OS X Tests.xcscheme]: Was this removed intentionally?

**Comment by bpoplauschi** [comment]
I noticed a CI build improvement (expected) with this PR. Looking at 2 random values, `master` took 19'37" and `lolgear:cleanup_of_tests` 15'28". We can't be sure this is the exact improvement (as conditions may vary), but definitely an improvement. On master, `pod setup` took 66 seconds, while `pod repo update` another 34 seconds.

**Comment by bpoplauschi** [comment]
I noticed a CI build improvement (expected) with this PR. Looking at 2 random values, `master` took 19'37" and `lolgear:cleanup_of_tests` 15'28". We can't be sure this is the exact improvement (as conditions may vary), but definitely an improvement. On master, `pod setup` took 66 seconds, while `pod repo update` another 34 seconds.

**Comment by bpoplauschi** [comment]
I noticed a CI build improvement (expected) with this PR. Looking at 2 random values, `master` took 19'37" and `lolgear:cleanup_of_tests` 15'28". We can't be sure this is the exact improvement (as conditions may vary), but definitely an improvement. On master, `pod setup` took 66 seconds, while `pod repo update` another 34 seconds.

**Comment by bpoplauschi** [comment]
I noticed a CI build improvement (expected) with this PR. Looking at 2 random values, `master` took 19'37" and `lolgear:cleanup_of_tests` 15'28". We can't be sure this is the exact improvement (as conditions may vary), but definitely an improvement. On master, `pod setup` took 66 seconds, while `pod repo update` another 34 seconds.

**Comment by bpoplauschi** [comment]
I noticed a CI build improvement (expected) with this PR. Looking at 2 random values, `master` took 19'37" and `lolgear:cleanup_of_tests` 15'28". We can't be sure this is the exact improvement (as conditions may vary), but definitely an improvement. On master, `pod setup` took 66 seconds, while `pod repo update` another 34 seconds.

**Comment by bpoplauschi** [comment]
@lolgear very interesting approach on mocking.
Curious why you choose `DDSMocking` instead of `DDMocking` or `DDMock`? :)
I will go ahead and merge, great job.
PS: didn't know about `__auto_type`.

**Comment by lolgear** [comment]
@bpoplauschi @ffried 
Only because of Smocking suit :)

Oh, I don't see your comments for awhile. Nice to know that this is problem of Github, lol :)

All stuff with deleting ( something ) from schemes - I could miss something. After project renaming old schemes are not valid anymore and I just hit "delete and autocreate" new.

**Comment by ffried** [comment]
Alright, I hope I got every duplicated comment now. Also, I guess there'll be inconsistencies depending on which DB server we get within the next two hours (according to status.github.com)...

**Comment by lolgear** [comment]
@bpoplauschi @ffried 
Only because of Smocking suit :)

Oh, I don't see your comments for awhile. Nice to know that this is problem of Github, lol :)

All stuff with deleting ( something ) from schemes - I could miss something. After project renaming old schemes are not valid anymore and I just hit "delete and autocreate" new.

**Comment by lolgear** [comment]
@bpoplauschi @ffried 
Only because of Smocking suit :)

Oh, I don't see your comments for awhile. Nice to know that this is problem of Github, lol :)

All stuff with deleting ( something ) from schemes - I could miss something. After project renaming old schemes are not valid anymore and I just hit "delete and autocreate" new.

**Pull request closed without merging**